model/src/main/scala/com/lynbrookrobotics/potassium/model/simulations/ui/SimulatorGUI.scala

```diff
@@ -0,0 +1,143 @@
+package com.lynbrookrobotics.potassium.model.simulations.ui
+
+import java.awt.event.{ActionEvent, ActionListener}
+import java.awt.{BorderLayout, Color, Dimension}
+import java.io.File
+import java.time.Clock
+import java.util
+import java.util.{Scanner, Timer, TimerTask}
+import javax.swing.event.ChangeEvent
+import javax.swing.{JButton, JFrame, JSlider}
+
+import com.lynbrookrobotics.potassium.units.Point
+import squants.space.{Angle, Degrees, Feet, Length}
+import squants.time.Seconds
+
+
+class SimulatorGUI(val fieldWidth: Length,
+                   val fieldHeight: Length,
+                   val boxSize: Int = 50,
+                   val maxSize: Dimension = new Dimension(600, 400)) {
+
+  /**
+    * This is the frame size, in pixels
+    */
+
+    val frameSize = new Dimension(
+    Math.min(fieldWidth.toFeet.toInt * boxSize, maxSize.width),
+    Math.min(fieldHeight.toFeet.toInt * boxSize, maxSize.height))
+
+  /**
+    * This is the Canvas (customized) which we draw with
+    */
+
+    val canvas = new Canvas(frameSize.width, frameSize.height)
+
+  /**
+    * This is the frame for our simulator
+    */
+
+    val frame = new JFrame("Simulator")
+    frame.setLayout(new BorderLayout())
+    frame.add(canvas)
+
+  /**
+    * This is your slider for our simulator
+    */
+
+    private val timeSlider = new JSlider()
+    private val layout = new BorderLayout()
+    frame.add(timeSlider)
+
+  /**
+    * This is the button for our simulator
+    */
+
+    val button = new JButton
+    frame.add(button)
+    button.setText("Play")
+    button.setSize(48,48)
+    button.setBackground(Color.GREEN)
+    button.setForeground(Color.RED)
+    button.setOpaque(true)
+
+  /**
+    * This is what the button does
+    */
+
+  def moveFunction(): Unit = {
+    timeSlider.setValue(timeSlider.getValue + 1)
```

Comment:
This seems arbitrary. You should be running the simulation in real time so the time passed when rendering matches the time passed in the log.